packages/core/src/signer/json-rpc-account.ts

```diff
@@ -0,0 +1,40 @@
+import { getAddress, type ByteArray, type Hex, type WalletClient } from "viem";
+import type { SmartAccountSigner } from "./types";
+import type { SignTypedDataParams } from "../account/types";
+
+export class JsonRpcAccountSigner implements SmartAccountSigner {
```

Comment:
nit:
```suggestion
export class WalletClientSigner implements SmartAccountSigner {
```

Reply:
Complying with this, since there is a `LocalAccountSigner`

![image](https://github.com/alchemyplatform/aa-sdk/assets/38040789/9ce97add-2632-4bab-8d14-8a40ddacd498)

Reply:
JsonRpcAccountSigner is too generic a name here. Viem's `WalletClient` isn't the only implementation of a JSON RPC provider, the `PublicClient` is as well. The ask here is to make the name specific to the use case. This signer is used to adapt a `WalletClient` to `SmartAccountSigner` so this should be a `WalletClientSigner` or some name that makes the use case more clear. 

`LocalAccountSigner` refers to the `viem` type [LocalAccount](https://github.com/wagmi-dev/viem/blob/cf4d5d8c666b7fd78a221974bb27162c0c6334b5/src/accounts/types.ts#L49). A `WalletClient` can actually be initialized with a `LocalAccount` as well as you pointed out. So it's not entirely accurate to call it just `JsonRpcAccountSigner`